src/zeta/aca_zeta_programming.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019 The Alcor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "aca_zeta_programming.h"
+#include "aca_ovs_l2_programmer.h"
+#include "aca_util.h"
+#include "aca_log.h"
+#include "aca_vlan_manager.h"
+#include "aca_oam_port_manager.h"
+
+using namespace aca_vlan_manager;
+using namespace aca_oam_port_manager;
+using namespace alcor::schema;
+namespace aca_zeta_programming
+{
+ACA_Zeta_Programming &ACA_Zeta_Programming::get_instance()
+{
+  static ACA_Zeta_Programming instance;
+  return instance;
+}
+
+int ACA_Zeta_Programming::create_or_update_zeta_config(const alcor::schema::AuxGateway current_AuxGateway,
+                                             const string vpc_id)
+{
+  zeta_config stZetaCfg;
+  int overall_rc = EXIT_SUCCESS;
+
+  stZetaCfg.group_id = current_AuxGateway.id();
+  for (auto destination : current_AuxGateway.destinations()) {
+    stZetaCfg.zeta_buckets.push_back(destination.ip_address());
+  }
+  uint32_t oam_server_port = current_AuxGateway.zeta_info().port_inband_operation();
+
+  uint32_t port;
+  ACA_Vlan_Manager::get_instance().get_oam_server_port(vpc_id, &port);
+  // oam_server_port is not set
+  if (port == 0) {
+    ACA_Vlan_Manager::get_instance().set_oam_server_port(vpc_id, oam_server_port);
+
+    //update oam_ports_table and add the OAM punt rule also if this is the first port in the VPC
+    Aca_Oam_Port_Manager::get_instance().add_vpc(oam_server_port, vpc_id);
```

Comment:
this implementation has two place to track similar things, please consolidate to avoid confusions:

set_oam_server_port - put the oam_port info into the vpc table
Aca_Oam_Port_Manager::get_instance().add_vpc - put vpc ids into the unordered_map using oam_port as the key.

Reply:
we really need to address this as we are using two places to store same information.